core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -671,7 +674,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
       classOf[LongWritable],
       classOf[BytesWritable],
       conf=conf)
-    val data = br.map{ case (k, v) => v.getBytes}
+    val data = br.map { case (k, v) =>
+      val bytes = v.getBytes
+      assert(bytes.length == recordLength, "Byte array does not have correct length")
+      bytes
```

Comment:
I meant should the user be told that the system can throw error when the records are not of the expected size. I dont have any strong feeling on this, just wondering.